Escape spaces found while parsing GPO XMLNS tags

willchenmark opened the pull request:
### What does this PR do?

### What issues does this PR fix or reference?
Escapes spaces found in XMLNS urls in Windows GPO documents with `%20`
Closes #66992

### Previous Behavior
Spaces would not be escaped, breaking lgpo state application.

### New Behavior
Spaces are properly escaped allow lgpo state application to operate correctly.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices, including the
[PR Guidelines](https://docs.saltproject.io/en/master/topics/development/pull_requests.html).

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

willchenmark committed:
escape spaces found while parsing GPO XMLNS tags

twangboy commented:
Let me know if you need help writing tests for this. Looks like a simple unit test on the `_encode_xmlns_url` function would be enough. And a changelog as well.

twangboy commented:
Also, let's make this fix on the 3006.x branch.

twangboy committed:
Add test for _encode_xmlns_url

twangboy committed:
Add changelog

twangboy commented:
OK, I added a test and created a change log. I also rebased on the 3006.x branch.

twangboy committed:
Fix pre-commit

willchenmark commented:
> OK, I added a test and created a change log. I also rebased on the 3006.x branch.

Thank you for this! I got pulled away and didn't have time to get back to this.
